p4-16/spec/P4-16-spec.mdk

```diff
@@ -5277,6 +5278,34 @@ of action `a_with_control_params`. Once the program is loaded,
 these entries are installed in the table in the order they are
 enumerated in the program.
 
+#### Size {#sec-size-table-property}
+
+The `size` is an optional property of a table.  When present, its
+value is always an integer.  It is specified in units of number of
+table entries.
+
+If a table has a `size` value specified for it with value `N`, it is
+recommended that a compiler should choose a data plane implementation
+that is capable of storing `N` table entries.  This does not guarantee
+that an _arbitrary_ set of `N` entries can always be inserted in such
+a table, only that there is _some_ set of `N` entries that can be
+inserted.  For example, attempts to add some combinations of `N`
+entries may fail because the compiler selected a hash table with
+`O(1)` guaranteed search time.  See "Size property of P4 tables and
+parser value sets"[^P4SizeProperty] for further discussion on some P4
+table implementations and what they are able to guarantee.
+
+[^P4SizeProperty]: <https://github.com/jafingerhut/p4-guide/blob/master/docs/p4-table-and-parser-value-set-sizes.md> TBD: please suggest a better place for such documents, perhaps a separate directory within <https://github.com/p4lang/p4-spec>
```

Comment:
Yes, we should use a more official URL. A file in that folder would be fine.

Reply:
I created a subdirectory p4-16/spec/docs/ and copied the file into there.